### ISSUE TITLE
Wrong error diagnostics message 'Cannot find module xxxx' 

### ISSUE THREAD

**Issue by mmorhun** [Microsoft/TypeScript#30422]
**TypeScript Version:**  3.4.0-dev.20190315

**Search Terms:**

`Cannot find module`, `wrong message`

**Steps to reproduce**

1. Create a new workspace (empty folder). Create subfolder `test-plugin` in it.

2. Install `npm install -g yo @theia/generator-plugin`

3. Generate a frontend plug-in: `yo @theia/plugin`. So, the folders structure should be as following:

```
ws-root
 |- test-plugin
    |- dist
    |- src
    |  |- test-plugin-frontend.ts
    |- node_modules   
    |  |- ...
    |- package.json
    |- tsconfig.json
    |- ...
```

4. Open `test-plugin-frontend.ts` and paste the following content:

```typescript
import * as theia from '@theia/plugin';

export function start(context: theia.PluginContext) {
    theia.window.onDidChangeVisibleTextEditors((editors: theia.TextEditor[]) => {
        let list: string = '';
        for (let editor of editors) {
            list += editor.document.uri.toString() + ', ';
        }
        console.log(list);
    });
  
   // theia.window.onDid
}

export function stop() { }
```
5. Build the project using `yarn` command

6. Uncomment the commented line. Change a few symbols and comment it again (Use `Ctrl + /`shortcut). Save the editor content.

**Expected behavior:**

No error diagnostic message, nothing has changed actually.

**Actual behavior:**

Error diagnostic message is appeared on the `@theia/plugin` import: `[typescript] Cannot find module '@theia/plugin'`

<img width="1615" alt="Screen Shot 2019-03-14 at 15 18 04" src="https://user-images.githubusercontent.com/3082655/54364022-65cf2f80-466c-11e9-848a-7dbe27960a7b.png">

**Comment by RyanCavanaugh** [comment]
Barring a more specific repro that doesn't involve a bunch of external command, this looks like a configuration error or a problem in another tool. Some way to produce this problem using just normal file operations would allow us to move forward.

**Comment by mmorhun** [comment]
I've created [repository](https://github.com/mmorhun/ts-30422) with source code for the problem.

### Steps to reproduce

1. Clone repository https://github.com/mmorhun/ts-30422
2. Edit the commented line in `src/test-plugin-frontend.ts`

Screencast:
![reproduce-ts-30422](https://user-images.githubusercontent.com/15607393/55383181-3b1c1c80-5530-11e9-84a9-8e942067f248.gif)
